homeassistant/components/tuya/alarm_control_panel.py

```diff
@@ -0,0 +1,139 @@
+"""Support for Tuya Alarm."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from tuya_iot import TuyaDevice, TuyaDeviceManager
+
+from homeassistant.components.alarm_control_panel import (
+    SUPPORT_ALARM_ARM_AWAY,
+    SUPPORT_ALARM_ARM_HOME,
+    SUPPORT_ALARM_TRIGGER,
+    AlarmControlPanelEntity,
+    AlarmControlPanelEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantTuyaData
+from .base import TuyaEntity
+from .const import DOMAIN, TUYA_DISCOVERY_NEW, DPCode
+
+ICON = "mdi:security"
+
+MODE_ALARM_ARM = "arm"
+MODE_ALARM_DISARMED = "disarmed"
+MODE_ALARM_HOME = "home"
+MODE_ALARM_SOS = "sos"
+
+
+@dataclass
+class TuyaAlarmControlPanelEntityDescription(AlarmControlPanelEntityDescription):
+    """Describe an Tuya Alarm entity."""
+
```

Comment:
This adds nothing, just use `AlarmControlPanelEntityDescription` directly instead